api/envoy/extensions/filters/network/tcp_proxy/v3/tcp_proxy.proto

```diff
@@ -84,6 +85,25 @@ message TcpProxy {
         [(validate.rules).repeated = {max_items: 1000}];
   }
 
+  message OnDemand {
```

Comment:
@krnowak does this API make sense to you? I'm wondering why TCP proxy needs to have a separate message like this to the HTTP case?

Reply:
I copied from @krnowak's PR. 
For this particular message, HCM, tcp proxy and other filters mostly likely should share.
If we put the name space and the location aside, is there anything specific to tcp_proxy concerning?

Reply:
No, I think mainly I was after just making this a common message.

Reply:
@htuch See https://github.com/envoyproxy/envoy/pull/20065/files#r823000906

I don't want to reuse the filters/http/on_demand/v3/OnDemand message there because that one carries the role of creating the C++ *OnDemand http filter*
Another reason is that that message is responsible for VHDS as well.

So we can either rename to OnDemandCds as the new message name to distinguish as best efforts, or
we can completely drop this message type, extract config.core.v3.ConfigSource odcds_config and timeout into TcpProxy config.

I prefer aggregate the ondemand configures in a dedicated message like this OnDemand but I am happy to dis-aggregate

Reply:
OK, that's a fair point.

Reply:
The on_demand extension indeed handles VHDS too, but before adding the ODCDS functionality to it, the message was empty. Maybe factoring the message out and putting it in the common namespace would be the way to go. Then the on_demand extension would just use it.